### DISCUSSION TITLE
This is how big the Re-Releases project folder is

### DISCUSSION THREAD

**Discussion by Blue-Shadow** [Realm667/Re-Releases#277] [General]
A few minutes ago, I pulled the recent changes. Needless to say, the process took about three minutes or so to complete, thanks to the fact that I have the repository on a thumb drive (external drives are slow) in addition to the size of the stuff pulled in. Then I thought: "how big has this project grown to in size?". I checked and this is what I got:

![fhffghsgh5](https://github.com/Realm667/Re-Releases/assets/4287687/ae91260f-70de-4c70-a831-dca3a340b6ef)

All I'm thinking right now is: why did Tormentor put all of these mods in the same repository?

**Comment by Tormentor667** [comment]
Well, when I first started with some of the Re-Releases, I didn't plan doing all of them. The whole thing has grown since then but it was never a problem for me tbh, as many of the projects share some data (yes, it's redundant, but for me this worked well). 

Do you suggest creating an own repository for each project?

**Comment by Blue-Shadow** [comment]
Unfortunately, for contributors such as myself, it's inconvenient. Currently, my interest lies in TCOTD2 only, so I don't need the other mods. If every mod was its own project, I'd cherry-pick and clone the ones I'm interested in.

Another issue is the commit history. Again, as I'm only interested in TCOTD2, naturally I'd like to see changes related to TCOTD2 only. As far as I know there isn't a way to filter that stuff.
 
> Do you suggest creating an own repository for each project?

It'd have been nice if this was done from the beginning. But, no, I'm not expecting you to do that. Consider it a lesson for the future, I guess.

**Comment by Tormentor667** [comment]
Yeah, I can understand that this is the case for you. For the other remasters, we have a certain set of developers (Ozymandias, Richard and me) who are working all on the same projects. But I still do get the point and I consider splitting it up. I just wonder how this could be done easily from now on.

**Comment by madame-rachelle** [comment]
The weakness with Git is that you pull all of the repository's history with every pull. That means if you have large files that have been redone several times (especially maps or images or music) that bloats the size of the pull.

The way to mitigate this is to do a shallow clone. This should ensure that you only get the data you need to delta between each pull, rather than the total delta of all changes with every commit. The way to do this is to specify `--depth 1` when cloning the repo.

**Comment by Tormentor667** [comment]
What's the con when doing this?

**Comment by madame-rachelle** [comment]
Your local repository can no longer be used as a backup if Github fails for whatever reason. You also won't be able to diff between commits as easily. You cannot checkout individual commits until they are fetched.

There's probably also a number of other issues - you'd have to work with a shallow repo to see them all. For everyday work and updates they're fine, for anything more advanced than that they're unsuitable.
